contracts/MoMABurn.sol

```diff
@@ -0,0 +1,136 @@
+    /// @param _tier1Address - new burn contract address for tier 1
+    /// @param _tier2Address - new burn contract address for tier 2
+    /// @param _tier3Address - new burn contract address for tier 3
+    function setMintContracts(
+        address _tier1Address,
+        address _tier2Address,
+        address _tier3Address
+    ) public onlyOwner {
+        require(
+            _tier1Address != address(0) &&
+                _tier2Address != address(0) &&
+                _tier3Address != address(0),
+            "Invalid contract address"
+        );
+        tier1Contract = FeralfileExhibitionV3_3(_tier1Address);
+        tier2Contract = FeralfileExhibitionV3_3(_tier2Address);
+        tier3Contract = FeralfileExhibitionV3_3(_tier3Address);
+    }
+
+    /// @notice setMintArtworkIDs use for update mint artwork ID
+    /// @param _tier1ArtworkID - new artwork ID for tier 1
+    /// @param _tier2ArtworkID - new artwork ID for tier 2
+    /// @param _tier3ArtworkID - new artwork ID for tier 3
+    function setMintArtworkIDs(
+        uint256 _tier1ArtworkID,
+        uint256 _tier2ArtworkID,
+        uint256 _tier3ArtworkID
+    ) public onlyOwner {
+        tier1ArtworkID = _tier1ArtworkID;
+        tier2ArtworkID = _tier2ArtworkID;
+        tier3ArtworkID = _tier3ArtworkID;
+    }
+
+    /// @notice setBurnEnabled use for update burn enable status
+    /// @param _isBurnEnabled - the condition is true or false
+    function setBurnEnabled(bool _isBurnEnabled) public onlyOwner {
+        isBurnEnabled = _isBurnEnabled;
+    }
+
+    /// @notice burnAndMint use for burn old editions and mint new edition
+    /// @param _burnedEditions - array of edition IDs to burn
+    function burnAndMint(uint256[] memory _burnedEditions) public {
+        require(isBurnEnabled, "Burn and mint status is not enabled");
+
+        require(
+            _burnedEditions.length == 3 ||
+                _burnedEditions.length == 6 ||
+                _burnedEditions.length == 9,
+            "Invalid number of burning editions"
+        );
+
+        bool isApprovalForAll = burnContract.isApprovedForAll(
+            _msgSender(),
+            address(this)
+        );
+
+        for (uint256 i = 0; i < _burnedEditions.length; i++) {
+            // Check duplicate with others
+            for (uint256 j = i + 1; j < _burnedEditions.length; j++) {
+                if (_burnedEditions[i] == _burnedEditions[j]) {
+                    revert("Invalid burning editions");
+                }
+            }
+
+            require(
+                burnContract.ownerOf(_burnedEditions[i]) == _msgSender(),
+                "Invalid owner of burned token"
+            );
+
+            if (!isApprovalForAll) {
+                require(
+                    burnContract.getApproved(_burnedEditions[i]) ==
+                        address(this),
+                    "Token need to approve for this contract"
+                );
+            }
+
```

Comment:
Missing check if the token we're going to burn here is actually belongs to the burning artwork. If someone accidentally submit another token which is not the 5k edition artwork but in same contract to this function we will burn it.

Reply:
Updated with MoMA artwork validation